src/sentry/api_gateway/api_gateway.py

```diff
@@ -1,31 +1,56 @@
 from __future__ import annotations
 
+from typing import Any, Callable
+
 from django.http.response import HttpResponseBase
 from rest_framework.request import Request
 
-from sentry.api_gateway.proxy import proxy_request
+from sentry.api_gateway.proxy import proxy_region_request, proxy_request
 from sentry.silo import SiloMode
+from sentry.silo.base import SiloLimit
+from sentry.types.region import find_all_multitenant_region_names, get_region_by_name
+
+REGION_PINNED_URLS = (
+    "/api/0/builtin-symbol-sources/",
+    "/api/0/grouping-configs/",
+)
 
 
-def _request_should_be_proxied(request: Request, view_func, view_kwargs) -> bool:
+def _get_view_silo_mode(view_func: Callable[..., HttpResponseBase]) -> frozenset[SiloMode] | None:
     view_class = getattr(view_func, "view_class", None)
-    current_silo_mode = SiloMode.get_current_mode()
-    if current_silo_mode == SiloMode.CONTROL and view_class is not None:
-        endpoint_silo_limit = getattr(view_class, "silo_limit", None)
-        if endpoint_silo_limit is not None:
-            endpoint_silo_set = endpoint_silo_limit.modes
-            return current_silo_mode not in endpoint_silo_set and "organization_slug" in view_kwargs
-    return False
+    if not view_class:
+        return None
+    if not hasattr(view_class, "silo_limit"):
+        return None
+    endpoint_silo_limit: SiloLimit = view_class.silo_limit
+    return endpoint_silo_limit.modes
 
 
-def proxy_request_if_needed(request: Request, view_func, view_kwargs) -> HttpResponseBase | None:
+def proxy_request_if_needed(
+    request: Request, view_func: Callable[..., HttpResponseBase], view_kwargs: dict[str, Any]
+) -> HttpResponseBase | None:
     """
-    Main execution flow for the API Gateway
-    returns None if proxying is not required
+    Main execution flow for the API Gateway.
+    returns None if proxying is not required, or a response if the proxy was successful.
     """
-    if not _request_should_be_proxied(request, view_func, view_kwargs):
+    current_silo_mode = SiloMode.get_current_mode()
+    if current_silo_mode != SiloMode.CONTROL:
         return None
 
-    # Request should be proxied at this point
-    org_slug = view_kwargs.get("organization_slug")
-    return proxy_request(request, org_slug)
+    silo_modes = _get_view_silo_mode(view_func)
+    if silo_modes and current_silo_mode in silo_modes:
+        return None
+
+    if "organization_slug" in view_kwargs:
+        org_slug = view_kwargs["organization_slug"]
+        return proxy_request(request, org_slug)
+
+    if request.path in REGION_PINNED_URLS:
+        # TODO(hybridcloud) This assumption may be unstable. We would need more metadata in regions
+        # to make this consistent and predictable.
+        regions = find_all_multitenant_region_names()
+        region = get_region_by_name(regions[0])
```

Comment:
We've been using `SENTRY_MONOLITH_REGION` for this purpose in other places and I think is a bit more accurate in the end.